plugins/obs-nvenc/nvenc.c

```diff
@@ -380,34 +382,47 @@ static bool init_encoder_base(struct nvenc_data *enc, obs_data_t *settings)
 	config->rcParams.qpMapMode = NV_ENC_QP_MAP_DELTA;
 
 	/* -------------------------- */
-	/* initialize                 */
-
-	info("settings:\n"
-	     "\tcodec:        %s\n"
-	     "\trate_control: %s\n"
-	     "\tbitrate:      %d\n"
-	     "\tmax_bitrate:  %d\n"
-	     "\tcq/cqp:       %ld\n"
-	     "\tkeyint:       %d\n"
-	     "\tpreset:       %s\n"
-	     "\ttuning:       %s\n"
-	     "\tmultipass:    %s\n"
-	     "\tprofile:      %s\n"
-	     "\twidth:        %d\n"
-	     "\theight:       %d\n"
-	     "\tb-frames:     %ld\n"
-	     "\tb-ref-mode:   %ld\n"
-	     "\tlookahead:    %s (%d)\n"
-	     "\taq:           %s\n"
-	     "\tsplit encode: %ld\n"
-	     "\tuser opts:    %s\n",
-	     get_codec_name(enc->codec), enc->props.rate_control, bitrate,
-	     max_bitrate, vbr ? enc->props.target_quality : enc->props.cqp,
-	     gop_size, enc->props.preset, enc->props.tune, enc->props.multipass,
-	     enc->props.profile, enc->cx, enc->cy, enc->props.bf,
-	     enc->props.bframe_ref_mode, lookahead ? "true" : "false",
-	     rc_lookahead, enc->props.adaptive_quantization ? "true" : "false",
-	     enc->props.split_encode, enc->props.opts_str);
+	/* log settings	              */
+
+	struct dstr log = {0};
+
+	dstr_catf(&log, "\tcodec:        %s\n", get_codec_name(enc->codec));
+	dstr_catf(&log, "\trate_control: %s\n", enc->props.rate_control);
+
+	if (bitrate)
+		dstr_catf(&log, "\tbitrate:      %d\n", bitrate);
+	if (vbr && !cqvbr)
+		dstr_catf(&log, "\tmax_bitrate:  %d\n", max_bitrate);
```

Comment:
I think Target Quality mode uses `maxBitrate` and doesn't use `bitrate` param, from looking at the [Programming Guide](https://docs.nvidia.com/video-technologies/video-codec-sdk/12.2/nvenc-video-encoder-api-prog-guide/index.html#rate-control). So the logging can be tweaked to match:

```suggestion
	if (bitrate && !cqvbr)
		dstr_catf(&log, "\tbitrate:      %d\n", bitrate);
	if (vbr)
		dstr_catf(&log, "\tmax_bitrate:  %d\n", max_bitrate);
```

Reply:
Ah yes minor brain fart here, fixed it now.